[WiP] Build LICENSE file dynamically

mistercrunch opened the pull request:
Some early work on license. Here's the output of the script that fetches license information from npm and organizes package by license:

Updated may 29th 2019
```
{
  "Custom: https://github.com/tmcw/jsonlint": [
    ""
  ],
  "MIT*": [
    "optimist",
    "exif-parser",
    "mapbox-gl",
    "expect.js",
    "split",
    "process",
    "trim",
    "typed-function"
  ],
  "Custom: https://github.com/IonicaBizau/node-blah": [
    "typpy",
    "flat-colors"
  ],
  "Custom: http://badges.github.io/stability-badges/dist/stable.svg": [
    "gl-vec2",
    "gl-mat3",
    "gl-vec3"
  ],
  "AFLv2.1": [],
  "Apache-2.0 WITH LLVM-exception": [
    "mousetrap"
  ],
  "CC0-1.0": [
    "string-hash"
  ],
  "Unlicense": [
    "tweetnacl"
  ]
}
```

mistercrunch committed:
Doing some work around LICENSEs

mistercrunch committed:
Dynamically building LICENSE.txt

mistercrunch committed:
lint

phoenix24 commented:
Just curious, why would one need to re-build license, especially in the light of these #5800 #6481?

mistercrunch commented:
This is about building the required `LICENSE` file at the root of the repo

bolkedebruin commented on scripts/liccheck.ini:
```diff
@@ -0,0 +1,25 @@
+        apache
+        apache 2.0
+        apache software license
+        gnu lgpl
```
No you cannot add these

mistercrunch committed:
remove *lgpl*

mistercrunch commented:
I updated the code, rebased, and regenerate the list of dubious packages in our JS tree:
```json
{'AFLv2.1': {'json-schema'},
 'Apache-2.0 WITH LLVM-exception': {'mousetrap'},
 'CC0-1.0': {'string-hash'},
 'Custom: http://badges.github.io/stability-badges/dist/stable.svg': {'gl-mat3',
                                                                      'gl-vec2',
                                                                      'gl-vec3'},
 'Custom: https://github.com/IonicaBizau/node-blah': {'flat-colors', 'typpy'},
 'Custom: https://github.com/tmcw/jsonlint': {''},
 'MIT*': {'exif-parser',
          'expect.js',
          'mapbox-gl',
          'optimist',
          'process',
          'split',
          'trim',
          'typed-function'},
 'Unlicense': {'tweetnacl'}}
```

mistercrunch commented:
Here's a dump of the production-only packages to figure out where those bad packages fit:
```
superset@0.999.0-dev /Users/maximebeauchemin/code/superset/superset/assets
├─┬ @data-ui/event-flow@0.0.54
│ ├─┬ @data-ui/forms@0.0.50
│ │ ├── prop-types@15.6.2 deduped
│ │ └── react-select@1.2.1 deduped
│ ├─┬ @data-ui/radial-chart@0.0.54
│ │ ├── @data-ui/shared@0.0.54 deduped
│ │ ├── @data-ui/theme@0.0.48
│ │ ├── @vx/event@0.0.140 deduped
│ │ ├── @vx/group@0.0.140 deduped
│ │ ├─┬ @vx/scale@0.0.140
│ │ │ └─┬ d3-scale@1.0.7
│ │ │   ├── d3-array@1.2.4 deduped
│ │ │   ├── d3-collection@1.0.7 deduped
│ │ │   ├── d3-color@1.2.3 deduped
│ │ │   ├── d3-format@1.3.2 deduped
│ │ │   ├── d3-interpolate@1.3.2 deduped
│ │ │   ├── d3-time@1.0.10 deduped
│ │ │   └── d3-time-format@2.1.3 deduped
│ │ ├── @vx/shape@0.0.140 deduped
│ │ ├── @vx/tooltip@0.0.140 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├── @data-ui/theme@0.0.48
│ ├─┬ @vx/axis@0.0.140
│ │ ├── @vx/group@0.0.140 deduped
│ │ ├── @vx/point@0.0.136 deduped
│ │ ├── @vx/shape@0.0.140 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └─┬ prop-types@15.5.10
│ │   ├── fbjs@0.8.17 deduped
│ │   └── loose-envify@1.4.0 deduped
│ ├─┬ @vx/bounds@0.0.140
│ │ └── prop-types@15.6.2 deduped
│ ├── @vx/clip-path@0.0.140
│ ├─┬ @vx/glyph@0.0.140
│ │ ├── @vx/group@0.0.140 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └─┬ d3-shape@1.2.2
│ │   └── d3-path@1.0.7 deduped
│ ├─┬ @vx/gradient@0.0.140
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/grid@0.0.140
│ │ ├── @vx/group@0.0.140 deduped
│ │ ├── @vx/point@0.0.136 deduped
│ │ ├── @vx/shape@0.0.140 deduped
│ │ └── classnames@2.2.6 deduped
│ ├─┬ @vx/group@0.0.140
│ │ └── classnames@2.2.6 deduped
│ ├─┬ @vx/legend@0.0.140
│ │ ├── @vx/group@0.0.140 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/pattern@0.0.140
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├── @vx/point@0.0.136
│ ├─┬ @vx/responsive@0.0.140
│ │ └── lodash@4.17.11 deduped
│ ├─┬ @vx/scale@0.0.140
│ │ └─┬ d3-scale@1.0.7
│ │   ├── d3-array@1.2.4 deduped
│ │   ├── d3-collection@1.0.7 deduped
│ │   ├── d3-color@1.2.3 deduped
│ │   ├── d3-format@1.3.2 deduped
│ │   ├── d3-interpolate@1.3.2 deduped
│ │   ├── d3-time@1.0.10 deduped
│ │   └── d3-time-format@2.1.3 deduped
│ ├─┬ @vx/shape@0.0.140
│ │ ├─┬ @vx/curve@0.0.140
│ │ │ └── d3-shape@1.2.2 deduped
│ │ ├── @vx/group@0.0.140 deduped
│ │ ├── @vx/point@0.0.136 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ ├── d3-shape@1.2.2 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/tooltip@0.0.140
│ │ ├── @vx/bounds@0.0.140 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ aphrodite@1.2.5
│ │ ├── asap@2.0.6 deduped
│ │ ├─┬ inline-style-prefixer@3.0.8
│ │ │ ├── bowser@1.9.4
│ │ │ └─┬ css-in-js-utils@2.0.1
│ │ │   ├── hyphenate-style-name@1.0.2
│ │ │   └── isobject@3.0.1
│ │ └── string-hash@1.1.3
│ ├── d3-array@1.2.4 deduped
│ ├── d3-format@1.3.2
│ ├── d3-selection@1.3.2 deduped
│ ├─┬ d3-zoom@1.7.3
│ │ ├── d3-dispatch@1.0.5 deduped
│ │ ├─┬ d3-drag@1.2.3
│ │ │ ├── d3-dispatch@1.0.5 deduped
│ │ │ └── d3-selection@1.3.2 deduped
│ │ ├── d3-interpolate@1.3.2 deduped
│ │ ├── d3-selection@1.3.2 deduped
│ │ └─┬ d3-transition@1.1.3
│ │   ├── d3-color@1.2.3 deduped
│ │   ├── d3-dispatch@1.0.5 deduped
│ │   ├── d3-ease@1.0.5
│ │   ├── d3-interpolate@1.3.2 deduped
│ │   ├── d3-selection@1.3.2 deduped
│ │   └── d3-timer@1.0.9 deduped
│ ├── immutable@3.8.2 deduped
│ ├── prop-types@15.6.2 deduped
│ ├── UNMET PEER DEPENDENCY react@^0.14.0 || ^15.0.0
│ ├── react-split-pane@0.1.85 deduped
│ ├─┬ react-with-styles@1.4.0
│ │ ├── deepmerge@1.5.2
│ │ ├─┬ global-cache@1.2.1
│ │ │ ├─┬ define-properties@1.1.3
│ │ │ │ └── object-keys@1.0.12
│ │ │ └─┬ is-symbol@1.0.2
│ │ │   └── has-symbols@1.0.0
│ │ ├── hoist-non-react-statics@1.2.0
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ react-with-styles-interface-aphrodite@1.2.0
│ │ ├── array-flatten@2.1.2
│ │ └─┬ has@1.0.3
│ │   └── function-bind@1.1.1
│ └─┬ recompose@0.23.5
│   ├── change-emitter@0.1.6
│   ├─┬ fbjs@0.8.17
│   │ ├── core-js@1.2.7
│   │ ├─┬ isomorphic-fetch@2.2.1
│   │ │ ├─┬ node-fetch@1.7.3
│   │ │ │ ├─┬ encoding@0.1.12
│   │ │ │ │ └─┬ iconv-lite@0.4.24
│   │ │ │ │   └── safer-buffer@2.1.2
│   │ │ │ └── is-stream@1.1.0
│   │ │ └── whatwg-fetch@3.0.0
│   │ ├── loose-envify@1.4.0 deduped
│   │ ├── object-assign@4.1.1 deduped
│   │ ├─┬ promise@7.3.1
│   │ │ └── asap@2.0.6 deduped
│   │ ├── setimmediate@1.0.5
│   │ └── ua-parser-js@0.7.19
│   ├── hoist-non-react-statics@1.2.0 deduped
│   └── symbol-observable@1.2.0 deduped
├─┬ @data-ui/histogram@0.0.64
│ ├─┬ @data-ui/shared@0.0.63
│ │ ├── @data-ui/theme@0.0.62 deduped
│ │ ├─┬ @vx/event@0.0.165
│ │ │ └── @vx/point@0.0.165
│ │ ├─┬ @vx/group@0.0.165
│ │ │ └── classnames@2.2.6 deduped
│ │ ├─┬ @vx/shape@0.0.168
│ │ │ ├─┬ @vx/curve@0.0.165
│ │ │ │ └── d3-shape@1.2.2 deduped
│ │ │ ├── @vx/group@0.0.165 deduped
│ │ │ ├── @vx/point@0.0.165 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ ├── d3-path@1.0.7 deduped
│ │ │ ├── d3-shape@1.2.2 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├─┬ @vx/tooltip@0.0.165
│ │ │ ├─┬ @vx/bounds@0.0.165
│ │ │ │ └── prop-types@15.6.2 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├── babel-runtime@6.26.0 deduped
│ │ ├── d3-array@1.2.4 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├── @data-ui/theme@0.0.62 deduped
│ ├── @vx/axis@0.0.140 deduped
│ ├─┬ @vx/curve@0.0.140
│ │ └── d3-shape@1.2.2 deduped
│ ├─┬ @vx/event@0.0.140
│ │ └── @vx/point@0.0.136 deduped
│ ├── @vx/glyph@0.0.140 deduped
│ ├── @vx/gradient@0.0.140 deduped
│ ├── @vx/group@0.0.140 deduped
│ ├── @vx/pattern@0.0.140 deduped
│ ├─┬ @vx/responsive@0.0.147
│ │ ├── lodash@4.17.11 deduped
│ │ └── resize-observer-polyfill@1.4.2
│ ├─┬ @vx/scale@0.0.140
│ │ └── d3-scale@1.0.7 deduped
│ ├── @vx/shape@0.0.140 deduped
│ ├── @vx/tooltip@0.0.140 deduped
│ ├─┬ babel-runtime@6.26.0
│ │ ├── core-js@2.6.0
│ │ └── regenerator-runtime@0.11.1
│ ├── d3-array@1.2.4 deduped
│ ├─┬ d3-scale@1.0.7
│ │ ├── d3-array@1.2.4 deduped
│ │ ├── d3-collection@1.0.7 deduped
│ │ ├── d3-color@1.2.3 deduped
│ │ ├── d3-format@1.3.2 deduped
│ │ ├── d3-interpolate@1.3.2 deduped
│ │ ├── d3-time@1.0.10 deduped
│ │ └── d3-time-format@2.1.3 deduped
│ ├── prop-types@15.6.2 deduped
│ └─┬ react-move@2.8.0
│   ├── d3-interpolate@1.3.2 deduped
│   └── d3-timer@1.0.9
├─┬ @data-ui/sparkline@0.0.54
│ ├─┬ @data-ui/shared@0.0.54
│ │ ├── @data-ui/theme@0.0.48
│ │ ├─┬ @vx/event@0.0.143
│ │ │ └── @vx/point@0.0.143
│ │ ├─┬ @vx/group@0.0.143
│ │ │ └── classnames@2.2.6 deduped
│ │ ├─┬ @vx/shape@0.0.145
│ │ │ ├─┬ @vx/curve@0.0.143
│ │ │ │ └── d3-shape@1.2.2 deduped
│ │ │ ├── @vx/group@0.0.143 deduped
│ │ │ ├── @vx/point@0.0.143 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ ├── d3-shape@1.2.2 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├─┬ @vx/tooltip@0.0.148
│ │ │ ├─┬ @vx/bounds@0.0.147
│ │ │ │ └── prop-types@15.6.2 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├── d3-array@1.2.4 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├── @data-ui/theme@0.0.8
│ ├── @vx/axis@0.0.140 deduped
│ ├─┬ @vx/curve@0.0.140
│ │ └── d3-shape@1.2.2 deduped
│ ├─┬ @vx/event@0.0.143
│ │ └── @vx/point@0.0.143
│ ├── @vx/glyph@0.0.140 deduped
│ ├── @vx/gradient@0.0.140 deduped
│ ├── @vx/group@0.0.140 deduped
│ ├── @vx/pattern@0.0.140 deduped
│ ├── @vx/point@0.0.136 deduped
│ ├─┬ @vx/responsive@0.0.140
│ │ └── lodash@4.17.11 deduped
│ ├─┬ @vx/scale@0.0.140
│ │ └─┬ d3-scale@1.0.7
│ │   ├── d3-array@1.2.4 deduped
│ │   ├── d3-collection@1.0.7 deduped
│ │   ├── d3-color@1.2.3 deduped
│ │   ├── d3-format@1.3.2 deduped
│ │   ├── d3-interpolate@1.3.2 deduped
│ │   ├── d3-time@1.0.10 deduped
│ │   └── d3-time-format@2.1.3 deduped
│ ├── @vx/shape@0.0.140 deduped
│ ├── d3-array@1.2.4 deduped
│ └── prop-types@15.6.2 deduped
├─┬ @data-ui/theme@0.0.62
│ └── babel-runtime@6.26.0 deduped
├─┬ @data-ui/xy-chart@0.0.61
│ ├─┬ @data-ui/shared@0.0.61
│ │ ├── @data-ui/theme@0.0.61 deduped
│ │ ├── @vx/event@0.0.165 deduped
│ │ ├── @vx/group@0.0.165 deduped
│ │ ├─┬ @vx/shape@0.0.168
│ │ │ ├── @vx/curve@0.0.165 deduped
│ │ │ ├── @vx/group@0.0.165 deduped
│ │ │ ├── @vx/point@0.0.165 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ ├── d3-path@1.0.7 deduped
│ │ │ ├── d3-shape@1.2.2 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├── @vx/tooltip@0.0.165 deduped
│ │ ├── babel-runtime@6.26.0 deduped
│ │ ├── d3-array@1.2.4 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @data-ui/theme@0.0.61
│ │ └── babel-runtime@6.26.0 deduped
│ ├─┬ @vx/axis@0.0.168
│ │ ├── @vx/group@0.0.165 deduped
│ │ ├── @vx/point@0.0.165 deduped
│ │ ├─┬ @vx/shape@0.0.168
│ │ │ ├── @vx/curve@0.0.165 deduped
│ │ │ ├── @vx/group@0.0.165 deduped
│ │ │ ├── @vx/point@0.0.165 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ ├── d3-path@1.0.7 deduped
│ │ │ ├── d3-shape@1.2.2 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├─┬ @vx/text@0.0.165
│ │ │ ├─┬ babel-plugin-lodash@3.3.4
│ │ │ │ ├─┬ @babel/helper-module-imports@7.0.0
│ │ │ │ │ └── @babel/types@7.2.2 deduped
│ │ │ │ ├─┬ @babel/types@7.2.2
│ │ │ │ │ ├── esutils@2.0.2 deduped
│ │ │ │ │ ├── lodash@4.17.11 deduped
│ │ │ │ │ └── to-fast-properties@2.0.0
│ │ │ │ ├─┬ glob@7.1.3
│ │ │ │ │ ├── fs.realpath@1.0.0
│ │ │ │ │ ├─┬ inflight@1.0.6
│ │ │ │ │ │ ├── once@1.4.0 deduped
│ │ │ │ │ │ └── wrappy@1.0.2
│ │ │ │ │ ├── inherits@2.0.3 deduped
│ │ │ │ │ ├─┬ minimatch@3.0.4
│ │ │ │ │ │ └─┬ brace-expansion@1.1.11
│ │ │ │ │ │   ├── balanced-match@1.0.0
│ │ │ │ │ │   └── concat-map@0.0.1
│ │ │ │ │ ├─┬ once@1.4.0
│ │ │ │ │ │ └── wrappy@1.0.2 deduped
│ │ │ │ │ └── path-is-absolute@1.0.1
│ │ │ │ ├── lodash@4.17.11 deduped
│ │ │ │ └── require-package-name@2.0.1
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ ├── lodash@4.17.11 deduped
│ │ │ └─┬ reduce-css-calc@1.3.0
│ │ │   ├── balanced-match@0.4.2
│ │ │   ├── math-expression-evaluator@1.2.17
│ │ │   └─┬ reduce-function-call@1.0.2
│ │ │     └── balanced-match@0.4.2
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/curve@0.0.165
│ │ └── d3-shape@1.2.2 deduped
│ ├─┬ @vx/event@0.0.165
│ │ └── @vx/point@0.0.165 deduped
│ ├─┬ @vx/glyph@0.0.165
│ │ ├── @vx/group@0.0.165 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └── d3-shape@1.2.2 deduped
│ ├─┬ @vx/gradient@0.0.165
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/grid@0.0.169
│ │ ├── @vx/group@0.0.165 deduped
│ │ ├── @vx/point@0.0.165 deduped
│ │ ├─┬ @vx/shape@0.0.168
│ │ │ ├── @vx/curve@0.0.165 deduped
│ │ │ ├── @vx/group@0.0.165 deduped
│ │ │ ├── @vx/point@0.0.165 deduped
│ │ │ ├── classnames@2.2.6 deduped
│ │ │ ├── d3-path@1.0.7 deduped
│ │ │ ├── d3-shape@1.2.2 deduped
│ │ │ └── prop-types@15.6.2 deduped
│ │ └── classnames@2.2.6 deduped
│ ├─┬ @vx/group@0.0.165
│ │ └── classnames@2.2.6 deduped
│ ├─┬ @vx/pattern@0.0.165
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├── @vx/point@0.0.165
│ ├─┬ @vx/responsive@0.0.165
│ │ ├── lodash@4.17.11 deduped
│ │ ├── prop-types@15.6.2 deduped
│ │ └── resize-observer-polyfill@1.5.0
│ ├── @vx/scale@0.0.165 deduped
│ ├─┬ @vx/shape@0.0.165
│ │ ├── @vx/curve@0.0.165 deduped
│ │ ├── @vx/group@0.0.165 deduped
│ │ ├── @vx/point@0.0.165 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ ├── d3-path@1.0.7
│ │ ├── d3-shape@1.2.2 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/stats@0.0.165
│ │ ├─┬ @vx/group@0.0.165
│ │ │ └── classnames@2.2.6 deduped
│ │ ├── @vx/scale@0.0.165 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └── d3-shape@1.2.2 deduped
│ ├─┬ @vx/tooltip@0.0.165
│ │ ├─┬ @vx/bounds@0.0.165
│ │ │ └── prop-types@15.6.2 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ └── prop-types@15.6.2 deduped
│ ├─┬ @vx/voronoi@0.0.165
│ │ ├─┬ @vx/group@0.0.165
│ │ │ └── classnames@2.2.6 deduped
│ │ ├── classnames@2.2.6 deduped
│ │ ├── d3-voronoi@1.1.4
│ │ └── prop-types@15.6.2 deduped
│ ├── babel-runtime@6.26.0 deduped
│ ├── d3-array@1.2.4 deduped
│ └── prop-types@15.6.2 deduped
├─┬ @superset-ui/chart@0.7.0
│ ├─┬ @superset-ui/core@0.3.0
│ │ └── lodash@4.17.11 deduped
│ └── reselect@4.0.0 deduped
├─┬ @superset-ui/color@0.7.0
│ ├── @superset-ui/core@0.7.0 deduped
│ └── d3-scale@2.1.2 deduped
├─┬ @superset-ui/connection@0.5.0
│ ├─┬ @babel/runtime@7.2.0
│ │ └── regenerator-runtime@0.12.1
│ └── whatwg-fetch@2.0.4
├─┬ @superset-ui/core@0.7.0
│ └── lodash@4.17.11 deduped
├─┬ @superset-ui/number-format@0.7.2
│ ├── @superset-ui/core@0.7.0 deduped
│ └── d3-format@1.3.2 deduped
├─┬ @superset-ui/time-format@0.7.2
│ ├── @superset-ui/core@0.7.0 deduped
│ ├── d3-time@1.0.10
│ └─┬ d3-time-format@2.1.3
│   └── d3-time@1.0.10 deduped
├─┬ @superset-ui/translation@0.7.0
│ └── jed@1.1.1
├─┬ @vx/legend@0.0.170
│ ├─┬ @vx/group@0.0.170
│ │ └── classnames@2.2.6 deduped
│ ├── classnames@2.2.6 deduped
│ └── prop-types@15.6.2 deduped
├─┬ @vx/responsive@0.0.172
│ ├── lodash@4.17.11 deduped
│ ├── prop-types@15.6.2 deduped
│ └── resize-observer-polyfill@1.5.0
├─┬ @vx/scale@0.0.165
│ └── d3-scale@2.1.2 deduped
├── abortcontroller-polyfill@1.2.1
├── bootstrap@3.4.0
├── bootstrap-slider@10.4.0
├── brace@0.11.1
├── classnames@2.2.6
├── d3@3.5.17
├── d3-array@1.2.4
├─┬ d3-cloud@1.2.5
│ └── d3-dispatch@1.0.5
├── d3-color@1.2.3
├── d3-hierarchy@1.1.8
├─┬ d3-sankey@0.4.2
│ ├── d3-array@1.2.4 deduped
│ ├── d3-collection@1.0.7
│ └─┬ d3-interpolate@1.3.2
│   └── d3-color@1.2.3 deduped
├─┬ d3-scale@2.1.2
│ ├── d3-array@1.2.4 deduped
│ ├── d3-collection@1.0.7 deduped
│ ├── d3-format@1.3.2 deduped
│ ├── d3-interpolate@1.3.2 deduped
│ ├── d3-time@1.0.10 deduped
│ └── d3-time-format@2.1.3 deduped
├── d3-selection@1.3.2
├── d3-svg-legend@1.13.0
├─┬ d3-tip@0.9.1
│ ├── d3-collection@1.0.7 deduped
│ └── d3-selection@1.3.2 deduped
├─┬ datamaps@0.5.9
│ ├── @types/d3@3.5.38
│ ├── d3@3.5.17 deduped
│ └─┬ topojson@1.6.27
│   ├── d3@3.5.17 deduped
│   ├─┬ d3-geo-projection@0.2.16
│   │ └── brfs@1.6.1 deduped
│   ├── d3-queue@2.0.3
│   ├─┬ optimist@0.3.7
│   │ └── wordwrap@0.0.3
│   ├── rw@1.3.3 deduped
│   └─┬ shapefile@0.3.1
│     ├── d3-queue@1.2.3
│     ├── iconv-lite@0.2.11
│     └── optimist@0.3.7 deduped
├─┬ datatables.net-bs@1.10.19
│ ├─┬ datatables.net@1.10.19
│ │ └── jquery@3.1.1 deduped
│ └── jquery@3.1.1 deduped
├─┬ deck.gl@5.3.5
│ ├─┬ @deck.gl/core@5.3.5
│ │ ├─┬ luma.gl@5.3.1
│ │ │ ├── math.gl@1.2.1 deduped
│ │ │ ├── probe.gl@1.0.4 deduped
│ │ │ ├── seer@0.2.4 deduped
│ │ │ └── webgl-debug@2.0.1
│ │ ├─┬ math.gl@1.2.1
│ │ │ ├── gl-mat3@1.0.0 deduped
│ │ │ ├── gl-mat4@1.2.0 deduped
│ │ │ ├── gl-quat@1.0.0 deduped
│ │ │ ├── gl-vec2@1.3.0 deduped
│ │ │ ├── gl-vec3@1.1.3 deduped
│ │ │ └── gl-vec4@1.0.1 deduped
│ │ ├── mjolnir.js@1.2.2 deduped
│ │ ├─┬ probe.gl@1.0.4
│ │ │ └─┬ asciify-image@0.1.3
│ │ │   ├─┬ couleurs@5.2.1
│ │ │   │ ├── flat-colors@3.0.0
│ │ │   │ ├── typpy@2.0.0
│ │ │   │ └── x256@0.0.2
│ │ │   ├─┬ jimp@0.2.28
│ │ │   │ ├── bignumber.js@2.4.0
│ │ │   │ ├── bmp-js@0.0.3
│ │ │   │ ├── es6-promise@3.3.1
│ │ │   │ ├── exif-parser@0.1.12
│ │ │   │ ├── file-type@3.9.0
│ │ │   │ ├── jpeg-js@0.2.0
│ │ │   │ ├─┬ load-bmfont@1.4.0
│ │ │   │ │ ├── buffer-equal@0.0.1 deduped
│ │ │   │ │ ├── mime@1.6.0 deduped
│ │ │   │ │ ├── parse-bmfont-ascii@1.0.6
│ │ │   │ │ ├── parse-bmfont-binary@1.0.6
│ │ │   │ │ ├─┬ parse-bmfont-xml@1.1.4
│ │ │   │ │ │ ├── xml-parse-from-string@1.0.1
│ │ │   │ │ │ └─┬ xml2js@0.4.19
│ │ │   │ │ │   ├── sax@1.2.4
│ │ │   │ │ │   └── xmlbuilder@9.0.7
│ │ │   │ │ ├── phin@2.9.3
│ │ │   │ │ ├─┬ xhr@2.5.0
│ │ │   │ │ │ ├── global@4.3.2 deduped
│ │ │   │ │ │ ├── is-function@1.0.1
│ │ │   │ │ │ ├─┬ parse-headers@2.0.1
│ │ │   │ │ │ │ ├─┬ for-each@0.3.3
│ │ │   │ │ │ │ │ └── is-callable@1.1.4
│ │ │   │ │ │ │ └── trim@0.0.1 deduped
│ │ │   │ │ │ └── xtend@4.0.1 deduped
│ │ │   │ │ └── xtend@4.0.1 deduped
│ │ │   │ ├── mime@1.6.0
│ │ │   │ ├─┬ mkdirp@0.5.1
│ │ │   │ │ └── minimist@0.0.8
│ │ │   │ ├─┬ pixelmatch@4.0.2
│ │ │   │ │ └── pngjs@3.3.3 deduped
│ │ │   │ ├── pngjs@3.3.3
│ │ │   │ ├── read-chunk@1.0.1
│ │ │   │ ├── request@2.88.0 deduped
│ │ │   │ ├─┬ stream-to-buffer@0.1.0
│ │ │   │ │ └── stream-to@0.2.2
│ │ │   │ ├── tinycolor2@1.4.1 deduped
│ │ │   │ └─┬ url-regex@3.2.0
│ │ │   │   └── ip-regex@1.0.3
│ │ │   ├── minimist@1.2.0 deduped
│ │ │   ├─┬ request@2.88.0
│ │ │   │ ├── aws-sign2@0.7.0
│ │ │   │ ├── aws4@1.8.0
│ │ │   │ ├── caseless@0.12.0
│ │ │   │ ├─┬ combined-stream@1.0.7
│ │ │   │ │ └── delayed-stream@1.0.0
│ │ │   │ ├── extend@3.0.2 deduped
│ │ │   │ ├── forever-agent@0.6.1
│ │ │   │ ├─┬ form-data@2.3.3
│ │ │   │ │ ├── asynckit@0.4.0
│ │ │   │ │ ├── combined-stream@1.0.7 deduped
│ │ │   │ │ └── mime-types@2.1.21 deduped
│ │ │   │ ├─┬ har-validator@5.1.3
│ │ │   │ │ ├─┬ ajv@6.6.1
│ │ │   │ │ │ ├── fast-deep-equal@2.0.1
│ │ │   │ │ │ ├── fast-json-stable-stringify@2.0.0
│ │ │   │ │ │ ├── json-schema-traverse@0.4.1
│ │ │   │ │ │ └─┬ uri-js@4.2.2
│ │ │   │ │ │   └── punycode@2.1.1
│ │ │   │ │ └── har-schema@2.0.0
│ │ │   │ ├─┬ http-signature@1.2.0
│ │ │   │ │ ├── assert-plus@1.0.0
│ │ │   │ │ ├─┬ jsprim@1.4.1
│ │ │   │ │ │ ├── assert-plus@1.0.0 deduped
│ │ │   │ │ │ ├── extsprintf@1.3.0
│ │ │   │ │ │ ├── json-schema@0.2.3
│ │ │   │ │ │ └─┬ verror@1.10.0
│ │ │   │ │ │   ├── assert-plus@1.0.0 deduped
│ │ │   │ │ │   ├── core-util-is@1.0.2 deduped
│ │ │   │ │ │   └── extsprintf@1.3.0 deduped
│ │ │   │ │ └─┬ sshpk@1.15.2
│ │ │   │ │   ├─┬ asn1@0.2.4
│ │ │   │ │   │ └── safer-buffer@2.1.2 deduped
│ │ │   │ │   ├── assert-plus@1.0.0 deduped
│ │ │   │ │   ├─┬ bcrypt-pbkdf@1.0.2
│ │ │   │ │   │ └── tweetnacl@0.14.5 deduped
│ │ │   │ │   ├─┬ dashdash@1.14.1
│ │ │   │ │   │ └── assert-plus@1.0.0 deduped
│ │ │   │ │   ├─┬ ecc-jsbn@0.1.2
│ │ │   │ │   │ ├── jsbn@0.1.1 deduped
│ │ │   │ │   │ └── safer-buffer@2.1.2 deduped
│ │ │   │ │   ├─┬ getpass@0.1.7
│ │ │   │ │   │ └── assert-plus@1.0.0 deduped
│ │ │   │ │   ├── jsbn@0.1.1
│ │ │   │ │   ├── safer-buffer@2.1.2 deduped
│ │ │   │ │   └── tweetnacl@0.14.5
│ │ │   │ ├── is-typedarray@1.0.0
│ │ │   │ ├── isstream@0.1.2
│ │ │   │ ├── json-stringify-safe@5.0.1
│ │ │   │ ├─┬ mime-types@2.1.21
│ │ │   │ │ └── mime-db@1.37.0
│ │ │   │ ├── oauth-sign@0.9.0
│ │ │   │ ├── performance-now@2.1.0 deduped
│ │ │   │ ├── qs@6.5.2
│ │ │   │ ├── safe-buffer@5.1.2 deduped
│ │ │   │ ├─┬ tough-cookie@2.4.3
│ │ │   │ │ ├── psl@1.1.31
│ │ │   │ │ └── punycode@1.4.1
│ │ │   │ ├─┬ tunnel-agent@0.6.0
│ │ │   │ │ └── safe-buffer@5.1.2 deduped
│ │ │   │ └── uuid@3.3.2
│ │ │   ├─┬ terminal-char-width@1.0.8
│ │ │   │ └── is-windows@0.1.1
│ │ │   └── window-size@0.1.4
│ │ ├── seer@0.2.4
│ │ └── viewport-mercator-project@5.3.0 deduped
│ ├─┬ @deck.gl/layers@5.3.5
│ │ ├── @deck.gl/core@5.3.5 deduped
│ │ ├── d3-hexbin@0.2.2
│ │ └── earcut@2.1.4 deduped
│ └─┬ @deck.gl/react@5.3.5
│   ├── @deck.gl/core@5.3.5 deduped
│   └── prop-types@15.6.2 deduped
├─┬ distributions@1.1.0
│ └── mathfn@1.1.0
├─┬ dnd-core@2.6.0
│ ├── asap@2.0.6
│ ├─┬ invariant@2.2.4
│ │ └── loose-envify@1.4.0 deduped
│ ├── lodash@4.17.11 deduped
│ └── redux@3.7.2 deduped
├── dompurify@1.0.8
├── geolib@2.0.24
├── immutable@3.8.2
├── jquery@3.1.1
├─┬ json-bigint@0.3.0
│ └── bignumber.js@7.2.1
├── lodash@4.17.11
├─┬ mapbox-gl@0.45.0
│ ├── @mapbox/gl-matrix@0.0.1
│ ├── @mapbox/jsonlint-lines-primitives@2.0.2
│ ├── @mapbox/mapbox-gl-supported@1.4.0
│ ├── @mapbox/point-geometry@0.1.0
│ ├── @mapbox/shelf-pack@3.2.0
│ ├── @mapbox/tiny-sdf@1.1.0
│ ├── @mapbox/unitbezier@0.0.0
│ ├─┬ @mapbox/vector-tile@1.3.1
│ │ └── @mapbox/point-geometry@0.1.0 deduped
│ ├── @mapbox/whoots-js@3.1.0
│ ├─┬ brfs@1.6.1
│ │ ├─┬ quote-stream@1.0.2
│ │ │ ├── buffer-equal@0.0.1
│ │ │ ├── minimist@1.2.0 deduped
│ │ │ └── through2@2.0.5 deduped
│ │ ├─┬ resolve@1.8.1
│ │ │ └── path-parse@1.0.6
│ │ ├─┬ static-module@2.2.5
│ │ │ ├── concat-stream@1.6.2 deduped
│ │ │ ├─┬ convert-source-map@1.6.0
│ │ │ │ └── safe-buffer@5.1.2 deduped
│ │ │ ├─┬ duplexer2@0.1.4
│ │ │ │ └── readable-stream@2.3.6 deduped
│ │ │ ├─┬ escodegen@1.9.1
│ │ │ │ ├── esprima@3.1.3
│ │ │ │ ├── estraverse@4.2.0
│ │ │ │ ├── esutils@2.0.2
│ │ │ │ ├─┬ optionator@0.8.2
│ │ │ │ │ ├── deep-is@0.1.3
│ │ │ │ │ ├── fast-levenshtein@2.0.6 deduped
│ │ │ │ │ ├─┬ levn@0.3.0
│ │ │ │ │ │ ├── prelude-ls@1.1.2 deduped
│ │ │ │ │ │ └── type-check@0.3.2 deduped
│ │ │ │ │ ├── prelude-ls@1.1.2
│ │ │ │ │ ├─┬ type-check@0.3.2
│ │ │ │ │ │ └── prelude-ls@1.1.2 deduped
│ │ │ │ │ └── wordwrap@1.0.0
│ │ │ │ └── source-map@0.6.1
│ │ │ ├─┬ falafel@2.1.0
│ │ │ │ ├── acorn@5.7.3
│ │ │ │ ├── foreach@2.0.5
│ │ │ │ ├── isarray@0.0.1
│ │ │ │ └── object-keys@1.0.12 deduped
│ │ │ ├── has@1.0.3 deduped
│ │ │ ├─┬ magic-string@0.22.5
│ │ │ │ └── vlq@0.2.3
│ │ │ ├─┬ merge-source-map@1.0.4
│ │ │ │ └── source-map@0.5.7
│ │ │ ├── object-inspect@1.4.1
│ │ │ ├── quote-stream@1.0.2 deduped
│ │ │ ├── readable-stream@2.3.6 deduped
│ │ │ ├── shallow-copy@0.0.1
│ │ │ ├─┬ static-eval@2.0.0
│ │ │ │ └── escodegen@1.9.1 deduped
│ │ │ └── through2@2.0.5 deduped
│ │ └── through2@2.0.5 deduped
│ ├── csscolorparser@1.0.3
│ ├── earcut@2.1.4
│ ├─┬ geojson-rewind@0.3.1
│ │ ├─┬ @mapbox/geojson-area@0.2.2
│ │ │ └── wgs84@0.0.0
│ │ ├─┬ concat-stream@1.6.2
│ │ │ ├── buffer-from@1.1.1
│ │ │ ├── inherits@2.0.3
│ │ │ ├── readable-stream@2.3.6 deduped
│ │ │ └── typedarray@0.0.6
│ │ ├── minimist@1.2.0
│ │ └─┬ sharkdown@0.1.0
│ │   ├─┬ cardinal@0.4.4
│ │   │ ├── ansicolors@0.2.1
│ │   │ └─┬ redeyed@0.4.4
│ │   │   └── esprima@1.0.4
│ │   ├── expect.js@0.2.0
│ │   ├── minimist@0.0.5
│ │   ├─┬ split@0.2.10
│ │   │ └── through@2.3.8 deduped
│ │   ├─┬ stream-spigot@2.1.2
│ │   │ └─┬ readable-stream@1.1.14
│ │   │   ├── core-util-is@1.0.2 deduped
│ │   │   ├── inherits@2.0.3 deduped
│ │   │   ├── isarray@0.0.1
│ │   │   └── string_decoder@0.10.31
│ │   └── through@2.3.8
│ ├── geojson-vt@3.2.1
│ ├─┬ gray-matter@3.1.1
│ │ ├─┬ extend-shallow@2.0.1
│ │ │ └── is-extendable@0.1.1
│ │ ├─┬ js-yaml@3.12.0
│ │ │ ├─┬ argparse@1.0.10
│ │ │ │ └── sprintf-js@1.0.3
│ │ │ └── esprima@4.0.1
│ │ ├── kind-of@5.1.0
│ │ └── strip-bom-string@1.0.0
│ ├── grid-index@1.0.0
│ ├── minimist@0.0.8
│ ├─┬ pbf@3.1.0
│ │ ├── ieee754@1.1.12
│ │ └─┬ resolve-protobuf-schema@2.1.0
│ │   └── protocol-buffers-schema@3.3.2
│ ├── quickselect@1.1.1
│ ├── rw@1.3.3
│ ├─┬ shuffle-seed@1.1.6
│ │ └── seedrandom@2.4.4
│ ├─┬ sort-object@0.3.2
│ │ ├── sort-asc@0.1.0
│ │ └── sort-desc@0.1.1
│ ├─┬ supercluster@2.3.0
│ │ └── kdbush@1.0.1
│ ├─┬ through2@2.0.5
│ │ ├─┬ readable-stream@2.3.6
│ │ │ ├── core-util-is@1.0.2
│ │ │ ├── inherits@2.0.3 deduped
│ │ │ ├── isarray@1.0.0
│ │ │ ├── process-nextick-args@2.0.0
│ │ │ ├── safe-buffer@5.1.2
│ │ │ ├─┬ string_decoder@1.1.1
│ │ │ │ └── safe-buffer@5.1.2 deduped
│ │ │ └── util-deprecate@1.0.2
│ │ └── xtend@4.0.1 deduped
│ ├── tinyqueue@1.2.3
│ └─┬ vt-pbf@3.1.1
│   ├── @mapbox/point-geometry@0.1.0 deduped
│   ├── @mapbox/vector-tile@1.3.1 deduped
│   └── pbf@3.1.0 deduped
├─┬ mathjs@3.20.2
│ ├── complex.js@2.0.4
│ ├── decimal.js@9.0.1
│ ├── escape-latex@1.2.0
│ ├── fraction.js@4.0.4
│ ├── javascript-natural-sort@0.7.1
│ ├── seed-random@2.2.0
│ ├── tiny-emitter@2.0.2
│ └── typed-function@0.10.7
├── moment@2.23.0
├── mousetrap@1.6.2
├── mustache@2.3.2
├── nvd3@1.8.6
├─┬ prop-types@15.6.2
│ ├─┬ loose-envify@1.4.0
│ │ └── js-tokens@4.0.0
│ └── object-assign@4.1.1
├── re-resizable@4.11.0
├─┬ react@16.6.3
│ ├── loose-envify@1.4.0 deduped
│ ├── object-assign@4.1.1 deduped
│ ├── prop-types@15.6.2 deduped
│ └─┬ scheduler@0.11.3
│   ├── loose-envify@1.4.0 deduped
│   └── object-assign@4.1.1 deduped
├─┬ react-ace@5.10.0
│ ├── brace@0.11.1 deduped
│ ├── lodash.get@4.4.2
│ ├── lodash.isequal@4.5.0
│ └── prop-types@15.6.2 deduped
├─┬ react-bootstrap@0.31.5
│ ├── babel-runtime@6.26.0 deduped
│ ├── classnames@2.2.6 deduped
│ ├─┬ dom-helpers@3.4.0
│ │ └── @babel/runtime@7.2.0 deduped
│ ├── invariant@2.2.4 deduped
│ ├── keycode@2.2.0
│ ├── prop-types@15.6.2 deduped
│ ├─┬ prop-types-extra@1.1.0
│ │ ├── react-is@16.6.3 deduped
│ │ └── warning@3.0.0 deduped
│ ├─┬ react-overlays@0.7.4
│ │ ├── classnames@2.2.6 deduped
│ │ ├── dom-helpers@3.4.0 deduped
│ │ ├── prop-types@15.6.2 deduped
│ │ ├── prop-types-extra@1.1.0 deduped
│ │ └── warning@3.0.0 deduped
│ ├─┬ uncontrollable@4.1.0
│ │ └── invariant@2.2.4 deduped
│ └─┬ warning@3.0.0
│   └── loose-envify@1.4.0 deduped
├── react-bootstrap-dialog@0.10.0
├─┬ react-bootstrap-slider@2.1.5
│ ├── bootstrap-slider@9.9.0
│ └── es6bindall@0.0.9
├─┬ react-color@2.14.1
│ ├── lodash@4.17.11 deduped
│ ├── material-colors@1.2.6
│ ├── prop-types@15.6.2 deduped
│ ├─┬ reactcss@1.2.3
│ │ └── lodash@4.17.11 deduped
│ └── tinycolor2@1.4.1
├─┬ react-datetime@2.16.3
│ ├─┬ create-react-class@15.6.3
│ │ ├── fbjs@0.8.17 deduped
│ │ ├── loose-envify@1.4.0 deduped
│ │ └── object-assign@4.1.1 deduped
│ ├── object-assign@3.0.0
│ ├── prop-types@15.6.2 deduped
│ └── react-onclickoutside@6.7.1
├─┬ react-dnd@2.6.0
│ ├── disposables@1.0.2
│ ├── dnd-core@2.6.0 deduped
│ ├── hoist-non-react-statics@2.5.5
│ ├── invariant@2.2.4 deduped
│ ├── lodash@4.17.11 deduped
│ └── prop-types@15.6.2 deduped
├─┬ react-dnd-html5-backend@2.6.0
│ └── lodash@4.17.11 deduped
├─┬ react-dom@16.6.3
│ ├── loose-envify@1.4.0 deduped
│ ├── object-assign@4.1.1 deduped
│ ├── prop-types@15.6.2 deduped
│ └── scheduler@0.11.3 deduped
├─┬ react-gravatar@2.6.3
│ ├── is-retina@1.0.3
│ ├─┬ md5@2.2.1
│ │ ├── charenc@0.0.2
│ │ ├── crypt@0.0.2
│ │ └── is-buffer@1.1.6
│ └─┬ query-string@4.3.4
│   ├── object-assign@4.1.1 deduped
│   └── strict-uri-encode@1.1.0
├─┬ react-hot-loader@4.6.0
│ ├── fast-levenshtein@2.0.6
│ ├─┬ global@4.3.2
│ │ ├─┬ min-document@2.19.0
│ │ │ └── dom-walk@0.1.1
│ │ └── process@0.5.2
│ ├── hoist-non-react-statics@2.5.5
│ ├─┬ loader-utils@1.1.0
│ │ ├── big.js@3.2.0
│ │ ├── emojis-list@2.1.0
│ │ └── json5@0.5.1
│ ├── lodash.merge@4.6.1
│ ├── prop-types@15.6.2 deduped
│ ├── react-lifecycles-compat@3.0.4
│ ├── shallowequal@1.1.0
│ └── source-map@0.7.3
├─┬ react-loadable@5.5.0
│ └── prop-types@15.6.2 deduped
├─┬ react-map-gl@3.3.9
│ ├── babel-runtime@6.26.0 deduped
│ ├── immutable@3.8.2 deduped
│ ├─┬ mapbox-gl@0.50.0
│ │ ├── @mapbox/geojson-types@1.0.2
│ │ ├── @mapbox/jsonlint-lines-primitives@2.0.2 deduped
│ │ ├── @mapbox/mapbox-gl-supported@1.4.0 deduped
│ │ ├── @mapbox/point-geometry@0.1.0 deduped
│ │ ├── @mapbox/tiny-sdf@1.1.0 deduped
│ │ ├── @mapbox/unitbezier@0.0.0 deduped
│ │ ├── @mapbox/vector-tile@1.3.1 deduped
│ │ ├── @mapbox/whoots-js@3.1.0 deduped
│ │ ├── brfs@1.6.1 deduped
│ │ ├── csscolorparser@1.0.3 deduped
│ │ ├── earcut@2.1.4 deduped
│ │ ├── geojson-rewind@0.3.1 deduped
│ │ ├── geojson-vt@3.2.1 deduped
│ │ ├── gl-matrix@2.8.1
│ │ ├── grid-index@1.0.0 deduped
│ │ ├── minimist@0.0.8
│ │ ├── murmurhash-js@1.0.0
│ │ ├── pbf@3.1.0 deduped
│ │ ├── potpack@1.0.1
│ │ ├── quickselect@1.1.1 deduped
│ │ ├── rw@1.3.3 deduped
│ │ ├── supercluster@4.1.1 deduped
│ │ ├── tinyqueue@1.2.3 deduped
│ │ └── vt-pbf@3.1.1 deduped
│ ├─┬ math.gl@2.2.0
│ │ ├── @babel/runtime@7.2.0 deduped
│ │ ├── gl-mat3@1.0.0
│ │ ├── gl-mat4@1.2.0
│ │ ├─┬ gl-quat@1.0.0
│ │ │ ├── gl-mat3@1.0.0 deduped
│ │ │ ├── gl-vec3@1.1.3 deduped
│ │ │ └── gl-vec4@1.0.1 deduped
│ │ ├── gl-vec2@1.3.0
│ │ ├── gl-vec3@1.1.3
│ │ └── gl-vec4@1.0.1
│ ├─┬ mjolnir.js@1.2.2
│ │ └── hammerjs@2.0.8
│ ├── prop-types@15.6.2 deduped
│ └─┬ viewport-mercator-project@6.1.0
│   ├── @babel/runtime@7.2.0 deduped
│   └── gl-matrix@3.0.0-0
├─┬ react-markdown@3.6.0
│ ├─┬ mdast-add-list-metadata@1.0.1
│ │ └── unist-util-visit-parents@1.1.2
│ ├── prop-types@15.6.2 deduped
│ ├─┬ remark-parse@5.0.0
│ │ ├── collapse-white-space@1.0.4
│ │ ├── is-alphabetical@1.0.2
│ │ ├── is-decimal@1.0.2
│ │ ├── is-whitespace-character@1.0.2
│ │ ├── is-word-character@1.0.2
│ │ ├── markdown-escapes@1.0.2
│ │ ├─┬ parse-entities@1.2.0
│ │ │ ├── character-entities@1.2.2
│ │ │ ├── character-entities-legacy@1.1.2
│ │ │ ├── character-reference-invalid@1.1.2
│ │ │ ├─┬ is-alphanumerical@1.0.2
│ │ │ │ ├── is-alphabetical@1.0.2 deduped
│ │ │ │ └── is-decimal@1.0.2 deduped
│ │ │ ├── is-decimal@1.0.2 deduped
│ │ │ └── is-hexadecimal@1.0.2
│ │ ├── repeat-string@1.6.1
│ │ ├── state-toggle@1.0.1
│ │ ├── trim@0.0.1
│ │ ├── trim-trailing-lines@1.1.1
│ │ ├─┬ unherit@1.1.1
│ │ │ ├── inherits@2.0.3 deduped
│ │ │ └── xtend@4.0.1 deduped
│ │ ├─┬ unist-util-remove-position@1.1.2
│ │ │ └── unist-util-visit@1.4.0 deduped
│ │ ├── vfile-location@2.0.4
│ │ └── xtend@4.0.1 deduped
│ ├─┬ unified@6.2.0
│ │ ├── bail@1.0.3
│ │ ├── extend@3.0.2
│ │ ├── is-plain-obj@1.1.0
│ │ ├── trough@1.0.3
│ │ ├─┬ vfile@2.3.0
│ │ │ ├── is-buffer@1.1.6 deduped
│ │ │ ├── replace-ext@1.0.0
│ │ │ ├── unist-util-stringify-position@1.1.2
│ │ │ └─┬ vfile-message@1.1.0
│ │ │   ├── @types/unist@2.0.2
│ │ │   └── unist-util-stringify-position@1.1.2 deduped
│ │ └── x-is-string@0.1.0
│ ├─┬ unist-util-visit@1.4.0
│ │ └─┬ unist-util-visit-parents@2.0.1
│ │   └── unist-util-is@2.1.2
│ └── xtend@4.0.1
├─┬ react-redux@5.1.1
│ ├── @babel/runtime@7.2.0 deduped
│ ├─┬ hoist-non-react-statics@3.2.1
│ │ └── react-is@16.6.3 deduped
│ ├── invariant@2.2.4 deduped
│ ├── loose-envify@1.4.0 deduped
│ ├── prop-types@15.6.2 deduped
│ ├── react-is@16.6.3
│ └── react-lifecycles-compat@3.0.4 deduped
├─┬ react-search-input@0.11.3
│ ├── fuse.js@3.3.0
│ └── prop-types@15.6.2 deduped
├─┬ react-select@1.2.1
│ ├── classnames@2.2.6 deduped
│ ├── prop-types@15.6.2 deduped
│ └─┬ react-input-autosize@2.2.1
│   └── prop-types@15.6.2 deduped
├─┬ react-select-fast-filter-options@0.2.3
│ └── js-search@1.4.2
├─┬ react-sortable-hoc@0.8.4
│ ├── babel-runtime@6.26.0 deduped
│ ├── invariant@2.2.4 deduped
│ └── prop-types@15.6.2 deduped
├─┬ react-split-pane@0.1.85
│ ├── prop-types@15.6.2 deduped
│ ├── react@16.6.3 deduped
│ ├── react-dom@16.6.3 deduped
│ ├── react-lifecycles-compat@3.0.4 deduped
│ └─┬ react-style-proptype@3.2.2
│   └── prop-types@15.6.2 deduped
├─┬ react-sticky@6.0.3
│ ├── prop-types@15.6.2 deduped
│ └─┬ raf@3.4.1
│   └── performance-now@2.1.0
├─┬ react-syntax-highlighter@7.0.4
│ ├── babel-runtime@6.26.0 deduped
│ ├── highlight.js@9.12.0
│ ├─┬ lowlight@1.9.2
│ │ ├─┬ fault@1.0.2
│ │ │ └── format@0.2.2
│ │ └── highlight.js@9.12.0 deduped
│ ├─┬ prismjs@1.15.0
│ │ └─┬ clipboard@2.0.4
│ │   ├─┬ good-listener@1.2.2
│ │   │ └── delegate@3.2.0
│ │   ├── select@1.1.2
│ │   └── tiny-emitter@2.0.2 deduped
│ └─┬ refractor@2.6.2
│   ├─┬ hastscript@5.0.0
│   │ ├─┬ comma-separated-tokens@1.0.5
│   │ │ └── trim@0.0.1 deduped
│   │ ├── hast-util-parse-selector@2.2.1
│   │ ├─┬ property-information@5.0.1
│   │ │ └── xtend@4.0.1 deduped
│   │ └─┬ space-separated-tokens@1.1.2
│   │   └── trim@0.0.1 deduped
│   ├── parse-entities@1.2.0 deduped
│   └── prismjs@1.15.0 deduped
├─┬ react-virtualized@9.19.1
│ ├── babel-runtime@6.26.0 deduped
│ ├── classnames@2.2.6 deduped
│ ├── dom-helpers@3.4.0 deduped
│ ├── loose-envify@1.4.0 deduped
│ ├── prop-types@15.6.2 deduped
│ └── react-lifecycles-compat@3.0.4 deduped
├─┬ react-virtualized-select@3.1.3
│ ├── babel-runtime@6.26.0 deduped
│ ├── prop-types@15.6.2 deduped
│ ├── react-select@1.2.1 deduped
│ └── react-virtualized@9.19.1 deduped
├── reactable-arc@0.14.42
├─┬ redux@3.7.2
│ ├── lodash@4.17.11 deduped
│ ├── lodash-es@4.17.11
│ ├── loose-envify@1.4.0 deduped
│ └── symbol-observable@1.2.0
├── redux-localstorage@0.4.1
├── redux-thunk@2.3.0
├── redux-undo@1.0.0-beta9-9-7
├── reselect@4.0.0
├─┬ shortid@2.2.14
│ └── nanoid@2.0.0
├── srcdoc-polyfill@1.0.0
├─┬ supercluster@4.1.1
│ └── kdbush@2.0.1
├── underscore@1.9.1
├── urijs@1.19.1
└─┬ viewport-mercator-project@5.3.0
  ├── @babel/runtime@7.2.0 deduped
  └─┬ math.gl@2.2.0
    ├── @babel/runtime@7.2.0 deduped
    ├── gl-mat3@1.0.0 deduped
    ├── gl-mat4@1.2.0 deduped
    ├── gl-quat@1.0.0 deduped
    ├── gl-vec2@1.3.0 deduped
    ├── gl-vec3@1.1.3 deduped
    └── gl-vec4@1.0.1 deduped

```

mistercrunch committed:
Improvements

mistercrunch commented:
The list without `deck.gl` (assuming we could make that an optional dep)
```
{
  "Custom: https://github.com/tmcw/jsonlint": [
    ""
  ],
  "MIT*": [
    "optimist",
    "process",
    "split",
    "trim",
    "mapbox-gl",
    "expect.js",
    "typed-function"
  ],
  "Apache-2.0 WITH LLVM-exception": [
    "mousetrap"
  ],
  "CC0-1.0": [
    "string-hash"
  ]
}
```

mistercrunch committed:
add to whitelist

mistercrunch commented:
No need for this anymore with FOSSA
